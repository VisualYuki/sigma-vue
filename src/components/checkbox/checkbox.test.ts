import {mount} from '@vue/test-utils'
import {describe, expect, it} from 'vitest'
import {UiCheckbox} from '.'

describe('', () => {
	it.each([
		['label prop', {props: {label: 'custom label'}}],
		['size prop', {props: {size: 'small'}}],
		['disabled', {props: {disabled: true}}]
	])('%s', (name, {props}) => {
		//@ts-ignore
		const wrapper = mount(UiCheckbox, {props})

		expect(wrapper.html()).toMatchSnapshot()
	})
})
