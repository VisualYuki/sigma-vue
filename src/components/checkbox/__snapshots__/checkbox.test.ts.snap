// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`disabled 1`] = `
"<div class="flex justify-start items-center cursor-not-allowed opacity-50">
  <div class="rounded-sm border-solid border border-(--color-neutral-500) w-4 h-4">
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`label prop 1`] = `
"<div class="cursor-pointer flex justify-start items-center">
  <div class="rounded-sm border-solid border border-(--color-neutral-500) w-4 h-4">
    <!--v-if-->
  </div>
  <div class="ms-2 pb-1 text-base">custom label</div>
</div>"
`;

exports[`size prop 1`] = `
"<div class="cursor-pointer flex justify-start items-center">
  <div class="rounded-sm border-solid border border-(--color-neutral-500) w-3.5 h-3.5">
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;
